4-12-rabota-s-tabbar

michurinn opened the pull request:
![image](https://user-images.githubusercontent.com/109435741/205896090-eb4d6634-8089-4c02-b00a-96cf96e200be.png)
![image](https://user-images.githubusercontent.com/109435741/205896121-7076b880-7468-420f-9764-175734d0c8c2.png)

michurinn committed:
File .gitignore changed , directory assets created

michurinn committed:
3.6 basics_of_flutter_homework_2 : changed pubspec

michurinn committed:
txt

michurinn committed:
homework4

michurinn committed:
4-2-Ychimsya_verstat

michurinn committed:
исправил

michurinn committed:
Добавлена документуия к классу, исправлены баги

michurinn committed:
Исправлены замечания

michurinn committed:
4-3-appbar

michurinn committed:
text & textstyle in AppBar

michurinn committed:
1.

michurinn committed:
2.

michurinn committed:
4

michurinn committed:
4

michurinn committed:
Исправлены замечания по комментариям

michurinn committed:
Внесены поправки по комментариям

michurinn committed:
Переделано создание списка елементво главного scr

michurinn committed:
1

michurinn committed:
4-8-1 Homework

michurinn committed:
1

michurinn committed:
Homework 4-10 Images

michurinn committed:
4-10 images

michurinn committed:
4-12-rabota_s_tabbar_bottomNavigationBar

internetova commented on lib/domain/sight.dart:
```diff
@@ -0,0 +1,15 @@
+  final String imagePath;
+  final String workTime;
+
+  const Sight(this.name, this.lat, this.lon, this.details, this.type,
```
параметры именованные лучше делать, особенно когда их много

internetova commented on lib/mocks.dart:
```diff
@@ -0,0 +1,17 @@
+//Заглушки с местами
+final List mocks = [
+  const Sight(AppStrings.sight0name, 59.87717719312971, 29.908010865005355,
+      AppStrings.sight0subscr, AppStrings.sightType0, 'assets/olgin.jpg',AppStrings.sight1workTime),
```
Читать содержимое списка сложно. Ставьте запятые и форматируйте код. Изображения из сети берем.

internetova commented on lib/mocks.dart:
```diff
@@ -0,0 +1,17 @@
+
+//Заглушки с местами
+final List mocks = [
+  const Sight(AppStrings.sight0name, 59.87717719312971, 29.908010865005355,
```
Ну тут зачем выносить в стринги данные? В стрингах только постоянные значения храним, а тут заполните тут все и с форматированием кода 🤗

internetova commented on lib/ui/screen/favorite_card.dart:
```diff
@@ -0,0 +1,53 @@
+import 'package:places/res/AppColors.dart';
+import 'package:places/res/AppTypography.dart';
+
+import 'sight_card.dart';
```
лучше все через package

internetova commented on lib/ui/screen/favorite_card.dart:
```diff
@@ -0,0 +1,53 @@
+      child: Row(
+        children: [
+          SvgPicture.asset(
+            'assets/icons/calendar.svg',
```
```dart
class AppAssets {
  static const favorite = 'res/icons/heart_icon.svg';
}

internetova commented on lib/ui/screen/sight_card.dart:
```diff
@@ -0,0 +1,81 @@
+                shape: const RoundedRectangleBorder(
+                    borderRadius: BorderRadius.all(
+                  Radius.circular(10),
+                )),
```
,

internetova commented on lib/ui/screen/sight_card.dart:
```diff
@@ -0,0 +1,81 @@
+                          sight.type,
+                          style: AppTypography.smallBoldwhite,
+                        ))
+                  ])),
```
,

internetova commented on lib/ui/screen/sight_card.dart:
```diff
@@ -0,0 +1,81 @@
+                                    style: AppTypography.simpleText),
+                                getPlannedText(),
+                                Text(sight.details, style: AppTypography.small)
+                              ])))
```
,,,,,,,,,,

internetova commented on lib/ui/screen/sight_card.dart:
```diff
@@ -0,0 +1,81 @@
+      width: 20,
+      height: 20,
+      child: SvgPicture.asset(
+        'assets/icons/like.svg',
```
```dart
class AppAssets {
  static const favorite = 'res/icons/heart_icon.svg';
}

internetova commented on lib/ui/screen/sight_card.dart:
```diff
@@ -0,0 +1,81 @@
+                              children: [
+                                Text(sight.name,
+                                    style: AppTypography.simpleText),
+                                getPlannedText(),
```
Методы или классы
Не очень хорошая идея с билд методами. Можно сделать карточки разных типов

Flutter Voronezh Meetup | Build methods vs Widget:
https://youtu.be/PNJDqt3Z_7Q

Сравнение виджетов и вспомогательных методов:
https://youtu.be/IOyq-eTRhvo

internetova commented on lib/ui/screen/sight_details.dart:
```diff
@@ -0,0 +1,129 @@
+                child: Padding(
+                  padding: const EdgeInsets.symmetric(
+                      vertical: 11, horizontal: 13.5),
+                  child: SvgPicture.asset('assets/icons/back_arrow.svg'),
```
👆

internetova commented on lib/ui/screen/sight_details.dart:
```diff
@@ -0,0 +1,129 @@
+                ),
+              ),
+            ),
+          ]),
```
,

internetova commented on lib/ui/screen/sight_details.dart:
```diff
@@ -0,0 +1,129 @@
+  final Sight sight;
+
+  @override
+  Widget build(BuildContext context) {
```
Очень длинный метод. было бы лучше разделить его на несколько смысловых классов.

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,74 @@
+          unselectedItemColor: AppColors.whiteSecondary,
+          items: const [
+            BottomNavigationBarItem(
+                icon: Icon(Icons.list_alt), label: "List of Places"),
```
кавычки почему то стали двойными 👀

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,74 @@
+                icon: Icon(Icons.favorite_rounded), label: "Favorite places"),
+            BottomNavigationBarItem(
+                icon: Icon(Icons.settings_outlined), label: "Settings"),
+          ]),
```
много где запятых не хватает

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,74 @@
+}
+
+// Создаёт Column из карт по списку mocks, разделённых SizedBox(height:20)
+class returnColumnFromList extends StatelessWidget {
```
почему название класса с маленькой буквы?

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,74 @@
+
+  @override
+  Widget build(BuildContext context) {
+    for (int i = 0; i < inputList.length; i++) {
```
Цикл внутри виджетов не очень хорошая идея.

internetova replied:
Не вижу улучшений 👀

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -0,0 +1,86 @@
+                ])));
+  }
+
+  TabBar _tabBar() {
```
в отдельный класс вынесите лучше

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -0,0 +1,86 @@
+        labelStyle: AppTypography.smallBoldwhite,
+        labelColor: AppColors.white,
+        labelPadding: EdgeInsets.symmetric(horizontal: 16, vertical: 10),
+        tabs: [Text("Хочу посетить"), Text("Посетил")]);
```
тексты вынести тоже в специальный класс - он у вас есть

internetova reviewed (commented):
Надо поработать над форматирование кода особенно, а также не забывать заглядывать в анализатор кода - он должен быть чистым.

michurinn commented:
Здравствуйте!
Мне не понятно как правильно ставить запятые: то у меня лишние, то много
где не хватает. Я  думал если компилятор не ругается значит всё нормально.
Где посмотреть на ету тему?



On Wed, Dec 7, 2022 at 5:45 PM Татьяна Сугина ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Надо поработать над форматирование кода особенно, а также не забывать
> заглядывать в анализатор кода - он должен быть чистым.
> ------------------------------
>
> In lib/domain/sight.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042271350>
> :
>
> > @@ -0,0 +1,15 @@
>
> +// Класс места, содержит информацию об интересном месте
>
> +import 'package:flutter/widgets.dart';
>
> +
>
> +class Sight {
>
> +  final String name;
>
> +  final double lat;
>
> +  final double lon;
>
> +  final String details;
>
> +  final String type;
>
> +  final String imagePath;
>
> +  final String workTime;
>
> +
>
> +  const Sight(this.name, this.lat, this.lon, this.details, this.type,
>
>
> параметры именованные лучше делать, особенно когда их много
> ------------------------------
>
> In lib/mocks.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042272720>
> :
>
> > @@ -0,0 +1,17 @@
>
> +import 'package:places/domain/sight.dart';
>
> +import 'package:places/res/AppStrings.dart';
>
> +
>
> +//Заглушки с местами
>
> +final List mocks = [
>
> +  const Sight(AppStrings.sight0name, 59.87717719312971, 29.908010865005355,
>
> +      AppStrings.sight0subscr, AppStrings.sightType0, 'assets/olgin.jpg',AppStrings.sight1workTime),
>
>
> Читать содержимое списка сложно. Ставьте запятые и форматируйте код.
> Изображения из сети берем.
> ------------------------------
>
> In lib/mocks.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042274849>
> :
>
> > @@ -0,0 +1,17 @@
>
> +import 'package:places/domain/sight.dart';
>
> +import 'package:places/res/AppStrings.dart';
>
> +
>
> +//Заглушки с местами
>
> +final List mocks = [
>
> +  const Sight(AppStrings.sight0name, 59.87717719312971, 29.908010865005355,
>
>
> Ну тут зачем выносить в стринги данные? В стрингах только постоянные
> значения храним, а тут заполните тут все и с форматированием кода 🤗
> ------------------------------
>
> In lib/ui/screen/favorite_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042275534>
> :
>
> > @@ -0,0 +1,53 @@
>
> +import 'package:flutter/material.dart';
>
> +import 'package:flutter/src/widgets/container.dart';
>
> +import 'package:flutter/src/widgets/framework.dart';
>
> +import 'package:flutter/widgets.dart';
>
> +import 'package:flutter_svg/flutter_svg.dart';
>
> +import 'package:places/domain/sight.dart';
>
> +import 'package:places/res/AppColors.dart';
>
> +import 'package:places/res/AppTypography.dart';
>
> +
>
> +import 'sight_card.dart';
>
>
> лучше все через package
> ------------------------------
>
> In lib/ui/screen/favorite_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042276803>
> :
>
> > +
>
> +  final Sight sight;
>
> +
>
> +  @override
>
> +  Widget build(BuildContext context) {
>
> +    return super.build(context);
>
> +  }
>
> +
>
> +  @override
>
> +  Widget getLeading() {
>
> +    return SizedBox(
>
> +      height: 20,
>
> +      child: Row(
>
> +        children: [
>
> +          SvgPicture.asset(
>
> +            'assets/icons/calendar.svg',
>
>
> class AppAssets {
>
>   static const favorite = 'res/icons/heart_icon.svg';
>
> }
>
> ------------------------------
>
> In lib/ui/screen/sight_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042278547>
> :
>
> > +class SightCard extends StatelessWidget {
>
> +  const SightCard({Key? key, required this.sight}) : super(key: key);
>
> +
>
> +  final Sight sight;
>
> +  @override
>
> +  Widget build(BuildContext context) {
>
> +    return Container(
>
> +        padding: const EdgeInsets.symmetric(horizontal: 16.0),
>
> +        child: AspectRatio(
>
> +            aspectRatio: 3 / 2,
>
> +            child: Card(
>
> +                color: AppColors.background,
>
> +                shape: const RoundedRectangleBorder(
>
> +                    borderRadius: BorderRadius.all(
>
> +                  Radius.circular(10),
>
> +                )),
>
>
> ,
> ------------------------------
>
> In lib/ui/screen/sight_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042278674>
> :
>
> > +                            topLeft: Radius.circular(10),
>
> +                            topRight: Radius.circular(10)),
>
> +                        image: DecorationImage(
>
> +                            fit: BoxFit.fitWidth,
>
> +                            image: AssetImage(sight.imagePath)),
>
> +                      ),
>
> +                    ),
>
> +                    Positioned(right: 18, top: 19, child: getLeading()),
>
> +                    Positioned(
>
> +                        left: 16,
>
> +                        top: 16,
>
> +                        child: Text(
>
> +                          sight.type,
>
> +                          style: AppTypography.smallBoldwhite,
>
> +                        ))
>
> +                  ])),
>
>
> ,
> ------------------------------
>
> In lib/ui/screen/sight_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042278838>
> :
>
> > +                          sight.type,
>
> +                          style: AppTypography.smallBoldwhite,
>
> +                        ))
>
> +                  ])),
>
> +                  Expanded(
>
> +                      child: Padding(
>
> +                          padding: const EdgeInsets.all(16.0),
>
> +                          child: Column(
>
> +                              mainAxisAlignment: getSpacesBehavior,
>
> +                              crossAxisAlignment: CrossAxisAlignment.stretch,
>
> +                              children: [
>
> +                                Text(sight.name,
>
> +                                    style: AppTypography.simpleText),
>
> +                                getPlannedText(),
>
> +                                Text(sight.details, style: AppTypography.small)
>
> +                              ])))
>
>
> ,,,,,,,,,,
> ------------------------------
>
> In lib/ui/screen/sight_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042279545>
> :
>
> > +
>
> +  // Чтобы переопределить в унаследованных классах
>
> +  MainAxisAlignment get getSpacesBehavior => MainAxisAlignment.spaceBetween;
>
> +
>
> +  // Чтобы переопределить в унаследованных классах
>
> +  Widget getPlannedText() {
>
> +    return const SizedBox.shrink();
>
> +  }
>
> +
>
> +  // Чтобы переопределить в унаследованных классах
>
> +  Widget getLeading() {
>
> +    return SizedBox(
>
> +      width: 20,
>
> +      height: 20,
>
> +      child: SvgPicture.asset(
>
> +        'assets/icons/like.svg',
>
>
> class AppAssets {
>
>   static const favorite = 'res/icons/heart_icon.svg';
>
> }
>
> ------------------------------
>
> In lib/ui/screen/sight_card.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042284625>
> :
>
> > +                        top: 16,
>
> +                        child: Text(
>
> +                          sight.type,
>
> +                          style: AppTypography.smallBoldwhite,
>
> +                        ))
>
> +                  ])),
>
> +                  Expanded(
>
> +                      child: Padding(
>
> +                          padding: const EdgeInsets.all(16.0),
>
> +                          child: Column(
>
> +                              mainAxisAlignment: getSpacesBehavior,
>
> +                              crossAxisAlignment: CrossAxisAlignment.stretch,
>
> +                              children: [
>
> +                                Text(sight.name,
>
> +                                    style: AppTypography.simpleText),
>
> +                                getPlannedText(),
>
>
> Методы или классы
> Не очень хорошая идея с билд методами. Можно сделать карточки разных типов
>
> Flutter Voronezh Meetup | Build methods vs Widget:
> https://youtu.be/PNJDqt3Z_7Q
>
> Сравнение виджетов и вспомогательных методов:
> https://youtu.be/IOyq-eTRhvo
> ------------------------------
>
> In lib/ui/screen/sight_details.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042285015>
> :
>
> > +          Stack(children: [
>
> +            SizedBox(
>
> +              height: 360,
>
> +              child: Image.asset(sight.imagePath, fit: BoxFit.fitHeight),
>
> +            ),
>
> +            Positioned(
>
> +              left: 16,
>
> +              top: 36,
>
> +              child: Container(
>
> +                width: 32,
>
> +                height: 32,
>
> +                color: Colors.white,
>
> +                child: Padding(
>
> +                  padding: const EdgeInsets.symmetric(
>
> +                      vertical: 11, horizontal: 13.5),
>
> +                  child: SvgPicture.asset('assets/icons/back_arrow.svg'),
>
>
> 👆
> ------------------------------
>
> In lib/ui/screen/sight_details.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042285329>
> :
>
> > +            ),
>
> +            Positioned(
>
> +              left: 16,
>
> +              top: 36,
>
> +              child: Container(
>
> +                width: 32,
>
> +                height: 32,
>
> +                color: Colors.white,
>
> +                child: Padding(
>
> +                  padding: const EdgeInsets.symmetric(
>
> +                      vertical: 11, horizontal: 13.5),
>
> +                  child: SvgPicture.asset('assets/icons/back_arrow.svg'),
>
> +                ),
>
> +              ),
>
> +            ),
>
> +          ]),
>
>
> ,
> ------------------------------
>
> In lib/ui/screen/sight_details.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042286834>
> :
>
> > @@ -0,0 +1,129 @@
>
> +import 'package:flutter/material.dart';
>
> +import 'package:flutter/src/widgets/framework.dart';
>
> +import 'package:flutter_svg/flutter_svg.dart';
>
> +import 'package:places/domain/sight.dart';
>
> +import 'package:places/res/AppTypography.dart';
>
> +import 'package:places/res/AppStrings.dart';
>
> +
>
> +class SideDetails extends StatelessWidget {
>
> +  const SideDetails({Key? key, required this.sight}) : super(key: key);
>
> +  final Sight sight;
>
> +
>
> +  @override
>
> +  Widget build(BuildContext context) {
>
>
> Очень длинный метод. было бы лучше разделить его на несколько смысловых
> классов.
> ------------------------------
>
> In lib/ui/screen/sight_list_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042288563>
> :
>
> > +            elevation: 0,
>
> +            title: const Text(
>
> +              AppStrings.listOfInterestingPlases,
>
> +              style: AppTypography.largeTitle,
>
> +            )),
>
> +      ),
>
> +      body:
>
> +          SingleChildScrollView(child: returnColumnFromList(inputList: mocks)),
>
> +      bottomNavigationBar: BottomNavigationBar(
>
> +          type: BottomNavigationBarType.fixed,
>
> +          showSelectedLabels: false,
>
> +          showUnselectedLabels: false,
>
> +          unselectedItemColor: AppColors.whiteSecondary,
>
> +          items: const [
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.list_alt), label: "List of Places"),
>
>
> кавычки почему то стали двойными 👀
> ------------------------------
>
> In lib/ui/screen/sight_list_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042288924>
> :
>
> > +          SingleChildScrollView(child: returnColumnFromList(inputList: mocks)),
>
> +      bottomNavigationBar: BottomNavigationBar(
>
> +          type: BottomNavigationBarType.fixed,
>
> +          showSelectedLabels: false,
>
> +          showUnselectedLabels: false,
>
> +          unselectedItemColor: AppColors.whiteSecondary,
>
> +          items: const [
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.list_alt), label: "List of Places"),
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.map_outlined), label: "Map"),
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.favorite_rounded), label: "Favorite places"),
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.settings_outlined), label: "Settings"),
>
> +          ]),
>
>
> много где запятых не хватает
> ------------------------------
>
> In lib/ui/screen/sight_list_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042289324>
> :
>
> > +          items: const [
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.list_alt), label: "List of Places"),
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.map_outlined), label: "Map"),
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.favorite_rounded), label: "Favorite places"),
>
> +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.settings_outlined), label: "Settings"),
>
> +          ]),
>
> +    );
>
> +  }
>
> +}
>
> +
>
> +// Создаёт Column из карт по списку mocks, разделённых SizedBox(height:20)
>
> +class returnColumnFromList extends StatelessWidget {
>
>
> почему название класса с маленькой буквы?
> ------------------------------
>
> In lib/ui/screen/sight_list_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042291001>
> :
>
> > +            BottomNavigationBarItem(
>
> +                icon: Icon(Icons.settings_outlined), label: "Settings"),
>
> +          ]),
>
> +    );
>
> +  }
>
> +}
>
> +
>
> +// Создаёт Column из карт по списку mocks, разделённых SizedBox(height:20)
>
> +class returnColumnFromList extends StatelessWidget {
>
> +  returnColumnFromList({super.key, required this.inputList});
>
> +  final List inputList;
>
> +  List<Widget> returnedColumn = [];
>
> +
>
> +  @override
>
> +  Widget build(BuildContext context) {
>
> +    for (int i = 0; i < inputList.length; i++) {
>
>
> Цикл внутри виджетов не очень хорошая идея.
> ------------------------------
>
> In lib/ui/screen/visiting_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042292317>
> :
>
> > +                showUnselectedLabels: false,
>
> +                unselectedItemColor: AppColors.whiteSecondary,
>
> +                items: [
>
> +                  const BottomNavigationBarItem(
>
> +                      icon: Icon(Icons.list_alt), label: "List of Places"),
>
> +                  const BottomNavigationBarItem(
>
> +                      icon: Icon(Icons.map_outlined), label: "Map"),
>
> +                  const BottomNavigationBarItem(
>
> +                      icon: Icon(Icons.favorite_rounded),
>
> +                      label: "Favorite places"),
>
> +                  const BottomNavigationBarItem(
>
> +                      icon: Icon(Icons.settings_outlined), label: "Settings")
>
> +                ])));
>
> +  }
>
> +
>
> +  TabBar _tabBar() {
>
>
> в отдельный класс вынесите лучше
> ------------------------------
>
> In lib/ui/screen/visiting_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1042292965>
> :
>
> > +                  const BottomNavigationBarItem(
>
> +                      icon: Icon(Icons.settings_outlined), label: "Settings")
>
> +                ])));
>
> +  }
>
> +
>
> +  TabBar _tabBar() {
>
> +    return const TabBar(
>
> +        indicator: BoxDecoration(
>
> +            color: AppColors.whiteSecondary,
>
> +            borderRadius: BorderRadius.all(Radius.circular(40.0))),
>
> +        unselectedLabelColor: AppColors.whiteSecondary2,
>
> +        unselectedLabelStyle: AppTypography.smallBoldwhite,
>
> +        labelStyle: AppTypography.smallBoldwhite,
>
> +        labelColor: AppColors.white,
>
> +        labelPadding: EdgeInsets.symmetric(horizontal: 16, vertical: 10),
>
> +        tabs: [Text("Хочу посетить"), Text("Посетил")]);
>
>
> тексты вынести тоже в специальный класс - он у вас есть
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#pullrequestreview-1208516870>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2C5WXJDQAIB6IYEWNGKHSDWMCPIBANCNFSM6AAAAAASVL4UDU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

michurinn committed:
Устранены замечания

internetova commented on lib/domain/sight.dart:
```diff
@@ -0,0 +1,20 @@
+      required this.details,
+      required this.type,
+      required this.imagePath,
+      required this.workTime});
```
Запятая - trailing comma:
https://docs.flutter.dev/development/tools/formatting

internetova commented:
> Здравствуйте! Мне не понятно как правильно ставить запятые: то у меня лишние, то много где не хватает. Я думал если компилятор не ругается значит всё нормально. Где посмотреть на ету тему?

@michurinn Про лишние я нигде не указывала 🤗
Запятая - trailing comma:
https://docs.flutter.dev/development/tools/formatting

internetova commented on lib/mocks.dart:
```diff
@@ -0,0 +1,31 @@
+      imagePath: 'http://s2.fotokto.ru/photo/full/219/2198428.jpg',
+      workTime: AppStrings.sight1workTime),
+  const Sight(
+      name: "Форт Константин",
```
К коду хорошо бы относиться аккуратнее - у вас то одинарные, то двойные кавычки.

internetova replied:
К коду хорошо бы относиться аккуратнее - у вас то одинарные, то двойные кавычки.

internetova commented on lib/ui/screen/favorite_card.dart:
```diff
@@ -0,0 +1,37 @@
+class FavoriteSight extends SightCard {
+  const FavoriteSight({super.key, required sight}) : super(sight: sight);
+
+
```
Не забывайте форматировать код. Есть комбинации клавиш для форматирования. Нажали бы - не было бы лишней пустой строки.

internetova commented on lib/ui/screen/favorite_card.dart:
```diff
@@ -0,0 +1,37 @@
+  }
+
+  @override
+  Widget getPlannedText() {
```
При написании виджетов такие приемы не используют. Я уже писала об этом. Сделайте две разные карточки - для списка мест и для избранного. В избранном можно добавить флаги для задания необходимых свойств когда они отличаются.

internetova commented on lib/ui/screen/sight_details.dart:
```diff
@@ -0,0 +1,36 @@
+    return Scaffold(
+      body: Column(
+        children: [
+          HeadWithImage(sight: sight),
```
Части виджета, если они больше нигде не используются можно оставить в этом же файле, только классы сделать приватными.

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,86 @@
+import 'package:places/mocks.dart';
+import 'package:places/res/app_colors.dart';
+import 'package:places/ui/screen/finished_card.dart';
+import '../../res/app_typography.dart';
```
package

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,86 @@
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: PreferredSize(
```
Аппбар лучше вынести в отдельный класс и разместить его можно в этом же файле.

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -0,0 +1,72 @@
+                      label: "Favorite places"),
+                  const BottomNavigationBarItem(
+                      icon: Icon(Icons.settings_outlined), label: "Settings")
+                ])));
```
Запятая - trailing comma:
https://docs.flutter.dev/development/tools/formatting

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -0,0 +1,72 @@
+                  FinishedSight(sight: mocks[1])
+                ],
+              )
+            ]),
```
,

internetova commented on lib/ui/screen/widgets/app_tab_bar.dart:
```diff
@@ -0,0 +1,19 @@
+      labelStyle: AppTypography.smallBoldwhite,
+      labelColor: AppColors.white,
+      labelPadding: EdgeInsets.symmetric(horizontal: 16, vertical: 10),
+      tabs: [Text(AppStrings.wouldLikeToVisit), Text(AppStrings.justVisited)]);
```
не хватает запятых после скобок

internetova commented on lib/ui/screen/widgets/app_tab_bar.dart:
```diff
@@ -0,0 +1,19 @@
+import 'package:places/res/app_strings.dart';
+import 'package:places/res/app_typography.dart';
+// ТабБАр приложения
+TabBar AppTabBar() {
```
он у вас к экрану Избранное относится, зачем выносить в отдельный файл?

internetova commented on lib/ui/screen/widgets/body_with_texts.dart:
```diff
@@ -0,0 +1,44 @@
+import 'package:places/domain/sight.dart';
+import 'package:places/res/app_typography.dart';
+// Тексты Название места и пр. для sight_details
+class BodyWithTexts extends StatelessWidget {
```
Лучше делать эти классы в родительском файле, т.к. они используются только там. И названия там делать с _ (приватными)

internetova commented on lib/ui/screen/widgets/head_with_image.dart:
```diff
@@ -0,0 +1,40 @@
+import 'package:places/res/app_assets.dart';
+
+// Картинка места и кнопка Назад для sight_details
+class HeadWithImage extends StatelessWidget {
```
Если уж вынесли в отдельные файлы части родительского экрана, то тогда уж лучше сделать папку Виджеты в папке с родительским экраном чтобы в общей папке с виджетами не лежало все подряд

internetova commented on pubspec.yaml:
```diff
@@ -1,89 +1,37 @@
+    - family: Roboto
+      fonts:
+        - asset: fonts/Roboto-Regular.ttf
+  
```
шрифты не все добавили что используются в приложении
https://docs.flutter.dev/development/tools/pubspec
```
fonts:
    - family: Roboto
      fonts:
        - asset: assets/fonts/Roboto-Thin.ttf
          weight: 100
```

internetova reviewed (commented):
Надо еще доработать

michurinn committed:
Исправлены замечания

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -59,28 +53,30 @@ class _SightListScreenState extends State<SightListScreen> {
   }
 }
+
+// Создаёт СПИСОК по карточкам mocks, разделённых SizedBox(height:20)
```
❗️Основные методы работы с коллекциями
https://youtu.be/W-gIKcS_hQA

метод map или expand

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -19,54 +18,82 @@ class _VisitingScreenState extends State<VisitingScreen> {
-                ])));
+            ],
+          )
+        ]),
```
👆,

internetova commented on lib/ui/screen/sight_list_screen.dart:
```diff
@@ -0,0 +1,82 @@
+
+// Создаёт СПИСОК по карточкам mocks, разделённых SizedBox(height:20)
+List<Widget> _returnColumnFromMocks = <Widget>[
+  mocks[0],
```
и у вас тут не виджет лежит, поэтому ошибка 🔴

internetova reviewed (commented):
1. Надо разобраться со свойствами таббара. Когда на него нажимаете вам нравится как он работает?

![2022-12-17 00 05 48](https://user-images.githubusercontent.com/39735343/208189028-25077672-c0be-466a-a03d-e67d3b880a95.gif)

А также по комментарию в коде исправить красный экран

michurinn committed:
Внесены правки по комментариям

michurinn commented:
Здравствуйте, я не понимаю что исправить в tabbar

сб, 17 дек. 2022 г., 00:07 Татьяна Сугина ***@***.***>:

> ***@***.**** commented on this pull request.
>
>    1. Надо разобраться со свойствами таббара. Когда на него нажимаете вам
>    нравится как он работает?
>
> [image: 2022-12-17 00 05 48]
> <https://user-images.githubusercontent.com/39735343/208189028-25077672-c0be-466a-a03d-e67d3b880a95.gif>
>
> А также по комментарию в коде исправить красный экран
> ------------------------------
>
> In lib/ui/screen/sight_list_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1051154726>
> :
>
> >    }
>
>  }
>
> +
>
> +// Создаёт СПИСОК по карточкам mocks, разделённых SizedBox(height:20)
>
>
> ❗️Основные методы работы с коллекциями
> https://youtu.be/W-gIKcS_hQA
>
> метод map или expand
> ------------------------------
>
> In lib/ui/screen/visiting_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1051155396>
> :
>
> > -                showUnselectedLabels: false,
>
> -                unselectedItemColor: AppColors.whiteSecondary,
>
> -                items: [
>
> -                  const BottomNavigationBarItem(
>
> -                      icon: Icon(Icons.list_alt), label: "List of Places"),
>
> -                  const BottomNavigationBarItem(
>
> -                      icon: Icon(Icons.map_outlined), label: "Map"),
>
> -                  const BottomNavigationBarItem(
>
> -                      icon: Icon(Icons.favorite_rounded),
>
> -                      label: "Favorite places"),
>
> -                  const BottomNavigationBarItem(
>
> -                      icon: Icon(Icons.settings_outlined), label: "Settings")
>
> -                ])));
>
> +            ],
>
> +          )
>
> +        ]),
>
>
> 👆,
> ------------------------------
>
> In lib/ui/screen/sight_list_screen.dart
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#discussion_r1051160299>
> :
>
> > +  Widget build(BuildContext context) {
>
> +    return AppBar(
>
> +      toolbarHeight: 120,
>
> +      backgroundColor: ThemeData.light().scaffoldBackgroundColor,
>
> +      elevation: 0,
>
> +      title: const Text(
>
> +        AppStrings.listOfInterestingPlases,
>
> +        style: AppTypography.largeTitle,
>
> +      ),
>
> +    );
>
> +  }
>
> +}
>
> +
>
> +// Создаёт СПИСОК по карточкам mocks, разделённых SizedBox(height:20)
>
> +List<Widget> _returnColumnFromMocks = <Widget>[
>
> +  mocks[0],
>
>
> и у вас тут не виджет лежит, поэтому ошибка 🔴
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/michurinn/surf-flutter-course-michurin/pull/12#pullrequestreview-1221456456>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2C5WXNSAHZA5OQOHLM6BT3WNTK2RANCNFSM6AAAAAASVL4UDU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>

michurinn committed:
Правки согласно комментариям

michurinn committed:
123

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -26,15 +26,16 @@ class _VisitingScreenState extends State<VisitingScreen> {
+              child: Material(
+                color: AppColors.background,
+                borderRadius: const BorderRadius.all(Radius.circular(40.0)),
+                child: _appTabBar(),
```
`_appTabBar()` надо вынести в отдельный класс, а не использовать билд метод

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -90,6 +91,7 @@ TabBar _appTabBar() {
     unselectedLabelStyle: AppTypography.smallBoldwhite,
     labelStyle: AppTypography.smallBoldwhite,
     labelColor: AppColors.white,
+    splashBorderRadius: BorderRadius.all(Radius.circular(40.0)),
```
Еще хорошо бы отключить эффект самого сплеша, т.к. выглядит сейчас лучше, но тоже не очень 🤗. Еще свойство есть для этого

internetova reviewed (commented):
Доделать таббар немного

michurinn committed:
Исправлены замечания по комментариям

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -26,37 +26,39 @@ class _VisitingScreenState extends State<VisitingScreen> {
           centerTitle: true,
           bottom: PreferredSize(
-            preferredSize: _appTabBar().preferredSize,
+            preferredSize: _AppTabBar().build(context).preferredSize,
```
Вам надо полностью аппбар сделать отдельным классом:
в строке 23 должно быть `appBar: FavoritesAppBar ...`
```dart
class FavoritesAppBar extends StatelessWidget implements PreferredSizeWidget 
```

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -26,37 +26,39 @@ class _VisitingScreenState extends State<VisitingScreen> {
           title: const Text("Избранное", style: AppTypography.subtitle),
```
Вынести тексты в класс со стрингами.

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -79,23 +81,27 @@ class _VisitingScreenState extends State<VisitingScreen> {
+        Text(AppStrings.wouldLikeToVisit),
+        Text(AppStrings.justVisited),
+      ],
+      splashFactory: NoSplash.splashFactory,
```
Этого не достататочно. Вы сами проверяли как работает?
Есть и другие свойства у виджета. Вам нужен `overlayColor`. Если провалиться в исходник, то там даже есть пример как использовать.
![2022-12-22 22 22 35](https://user-images.githubusercontent.com/39735343/209211679-5d65c004-49b2-4faa-a639-151299421162.gif)

internetova reviewed (commented):
Надо переделать аппбар в экране Избранное, а также пройти проверить по всему коду, что вы вынесли в се тексты в класс со стрингами.

michurinn committed:
Исправлены замечания по комментариям

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -80,12 +64,41 @@ class _VisitingScreenState extends State<VisitingScreen> {
+          const Text(AppStrings.favoritePlaces, style: AppTypography.subtitle),
+      centerTitle: true,
+      bottom: PreferredSize(
+        preferredSize: _AppTabBar().build(context).preferredSize,
```
Тут такие сложности ни к чему. Лучше так `Size.fromHeight(60)`

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -80,12 +64,41 @@ class _VisitingScreenState extends State<VisitingScreen> {
+          child: Material(
+            color: AppColors.background,
+            borderRadius: const BorderRadius.all(Radius.circular(40.0)),
+            child: _AppTabBar(),
```
Раз вынесли арбар в отдельный класс, то такая реализация не нужна. Лучше перенести сюда таббар со строки 100

internetova commented on lib/ui/screen/visiting_screen.dart:
```diff
@@ -80,12 +64,41 @@ class _VisitingScreenState extends State<VisitingScreen> {
   }
 }
 
+class FavoriteAppBar extends StatelessWidget implements PreferredSizeWidget {
```
Если у вас класс в файле с другим родительским классом и нигде не используется больше, делаем его приватным `class _FavoriteAppBar `

internetova reviewed (commented):
Еще немного хорошо бы поправить по комментариям 🤗

michurinn committed:
Исправлены замечания по комментариям
